native/cocos/base/Macros.h

```diff
@@ -409,3 +409,7 @@ It should work same as apples CFSwapInt32LittleToHost(..)
         #define CC_FORCE_INLINE inline
     #endif
 #endif
+
+// Macro definition to check if a type has padding
+#define CC_CHECK_STRUCT_NO_PADDING(T) \
+    static_assert(std::has_unique_object_representations_v<T>, "ERROR: Type has padding bytes!")
```

Comment:
Can not use `std::has_unique_object_representations_v<T>` to check padding, as in the following codes, it also returns false
```c++
struct D {
    int* ptr;
};

struct E {
    int id;
    std::string name;
    double* data;
};
```

Will return false.

Reply:
It could work for primitve types, and all gfx structs that need the added padding manually could work by `std:: has_unique_object_representations_v `

So do we need to consider stl containers, like `string`, `vector`, `list`, `map` in struct? 
And if so , any suggestion for how to check those types?

Reply:
The example you wrote:
```cpp
struct D {
    int* ptr;
};
```
returns true.

You could paste code 
```cpp
#include <type_traits>
#include <stdint.h>
#include <iostream>
#include <cstddef> // 包含 offsetof 宏

// 宏定义，检查类型是否有 padding
#define CHECK_NO_PADDING(T) \
    static_assert(std::has_unique_object_representations_v<T>, "Type " #T " has padding bytes!")


struct D {
    int* ptr;
};

struct E {
    int id;
    std::string name;
    double* data;
};

int main() {

  
  CHECK_NO_PADDING(D);
  CHECK_NO_PADDING(E);
   

    return 0;
}

```

in [cpp.sh](https://cpp.sh)  to test.

Reply:
> So do we need to consider stl containers, like string, vector, list, map in struct?
And if so , any suggestion for how to check those types?

It depends on the usage. But as this function is defined in Macros.h, so should add comments to say that it only works for primitive types.

Reply:
I will add a comment later.